pkg/gitprovider/bitbucket/bitbucket.go

```diff
@@ -289,6 +296,66 @@ func (p *provider) ListPullRequests(
 	return prs, nil
 }
 
+// MergePullRequest implements gitprovider.Interface.
+func (p *provider) MergePullRequest(
+	_ context.Context,
+	id int64,
+) (*gitprovider.PullRequest, bool, error) {
+	// Get the current PR to check its state
+	prOpts := &bitbucket.PullRequestsOptions{
+		Owner:    p.owner,
+		RepoSlug: p.repoSlug,
+		ID:       strconv.FormatInt(id, 10),
+	}
+
+	currentPRResp, err := p.client.GetPullRequest(prOpts)
+	if err != nil {
+		return nil, false, fmt.Errorf("error getting pull request %d: %w", id, err)
+	}
+
+	currentPR, err := toBitbucketPR(currentPRResp)
+	if err != nil {
+		return nil, false, fmt.Errorf("error parsing pull request response: %w", err)
+	}
+
+	// Check if PR is already merged
+	if currentPR.State == prStateMerged {
+		return toProviderPR(currentPR, currentPRResp), true, nil
+	}
+
+	// Check if PR is closed/declined
+	if currentPR.State == prStateDeclined || currentPR.State == prStateSuperseded {
+		return nil, false, fmt.Errorf("pull request %d is closed but not merged (state: %s)", id, currentPR.State)
+	}
```

Comment:
I'm not positive, but I believe Bitbucket Server/Data Center also has `CLOSED`...

But instead of checking for `DECLINED`, `SUPERSEDED`, and `CLOSED`, maybe it's better to just check for _not_ `OPEN`?

Reply:
I can't find `CLOSED` PR state on their documentation. I do see that declined and superseded is referred as closed in bitbucket.